de1plus/bluetooth.tcl

```diff
@@ -682,6 +682,40 @@ proc decentscale_tare {} {
 	userdata_append "decentscale : tare" [list ble write $::de1(scale_device_handle) $::de1(suuid_decentscale) $::sinstance($::de1(suuid_decentscale)) $::de1(cuuid_decentscale_write) $::cinstance($::de1(cuuid_decentscale_write)) $tare] 0
 }
 
+### Meater
+
+proc meater_enable_notifications {} {
+	if {$::de1(thermometer_device_handle) == 0 || $::settings(thermometer_bluetooth_type) != "apptionmeater"} {
+		return
+	}
+
+	if {[ifexists ::sinstance($::de1(suuid_decentscale))] == ""} {
+		msg "meater not connected, cannot enable weight notifications"
+		return
+	}
+
+	userdata_append "enable meater notifications" [list ble enable $::de1(thermometer_device_handle) $::de1(suuid_meater) $::sinstance($::de1(suuid_meater)) $::de1(cuuid_meater) $::cinstance($::de1(cuuid_meater))] 1
+}
+
+proc meater_parse_response { value } {
+	binary scan $value a2 raw
+	set temp [expr $raw / 16]
+
+	msg "Meater temperature is $temp"
+	process_temperature_value $temp
+}
+
+
+### MISC 
+
+proc process_temperature_value {value} {
+	if {$::de1(substate) == $::de1_substate_types_reversed("Steaming")} {
+		if {$value >= $::settings(target_milk_temperature)} {
+			start_idle
```

Comment:
To help prevent puzzles
`msg -NOTICE [format "Steaming stopped for temperature at %.1f for target %.1f" $value $::settings(target_milk_temperature)`